Infrastructure: Trilinos shipping with Ubuntu 18.04 / 20.04 and Debian bullseye (stable) no longer usable

Since tonight some of our nightly-tests using the official Docker image are failing. We have following lines in our CMakeLists.txt file: https://github.com/kronbichler/adaflo/blob/cf7636e90cd530fb5b19985331101f78773d71fd/CMakeLists.txt#L26-L54

During configuration it complains that Trilinos is missing: 
```
-- Configuring incomplete, errors occurred!
adaflo requires a deal.II installation built with support for Trilinos and p4est but one or both of these appears to be missing!

CMake Error at CMakeLists.txt:42 (MESSAGE):
  

  -- deal.II was built without support for Trilinos!
```

see https://github.com/kronbichler/adaflo/actions/runs/3835904943/jobs/6529539908.

@tamiko Do you have a clue? Locally, I cannot replicate the issue. Is it related to #14634?

FYI @mschreter

I also experience a problem when building a user project based off a custom deal.II docker instance (basically master with the commit frozen) that itself is `FROM dealii/dependencies:focal`. If continues with the bundled Kokkos (given that an acceptable Trilinos is no longer detected):
```
// detailed.log
#        DEAL_II_BUNDLED_INCLUDE_DIRS: /usr/src/dealii/bundled/kokkos-3.7.00/algorithms/src;/usr/src/dealii/bundled/kokkos-3.7.00/containers/src;/usr/src/dealii/bundled/kokkos-3.7.00/core/src;/usr/src/dealii/bundled/kokkos-3.7.00/simd/src;/usr/src/dealii/bundled/kokkos-3.7.00/tpls/desul/include
...
#      ( DEAL_II_WITH_TRILINOS = OFF )
```
then the build of deal.II is fine. The issue arises when finally building the user project:
```
/usr/local/include/deal.II/base/numbers.h:28:10: fatal error: Kokkos_Macros.hpp: No such file or directory
   28 | #include <Kokkos_Macros.hpp>
      |          ^~~~~~~~~~~~~~~~~~~
```
Are the `DEAL_II_BUNDLED_INCLUDE_DIRS` being propagated downstream?

--- a/cmake/configure/configure_20_trilinos.cmake
+++ b/cmake/configure/configure_20_trilinos.cmake
@@ -198,18 +198,18 @@ macro(feature_trilinos_find_external var)
       endif()
 
       #
-      # We require at least Trilinos 13.2
+      # We require at least Trilinos 12.14.1
       #
-      if(TRILINOS_VERSION VERSION_LESS 13.2)
+      if(TRILINOS_VERSION VERSION_LESS 12.14.1)
         message(STATUS "Could not find a sufficient Trilinos installation: "
-          "deal.II requires at least version 13.2 if the Trilinos installation includes Kokkos, "
+          "deal.II requires at least version 12.14.1 if the Trilinos installation includes Kokkos, "
           "but version ${TRILINOS_VERSION} was found."
           )
         set(TRILINOS_ADDITIONAL_ERROR_STRING
           ${TRILINOS_ADDITIONAL_ERROR_STRING}
           "The Trilinos installation (found at \"${TRILINOS_DIR}\")\n"
           "with version ${TRILINOS_VERSION} is too old.\n"
-          "deal.II requires at least version 13.2 if the Trilinos installation includes Kokkos.\n\n"
+          "deal.II requires at least version 12.14.1 if the Trilinos installation includes Kokkos.\n\n"
           )
         set(${var} FALSE)
       endif()

--- a/doc/external-libs/trilinos.html
+++ b/doc/external-libs/trilinos.html
@@ -50,7 +50,7 @@
 
     <p style="color: red">
       Note: The current version of deal.II requires at least Trilinos 12.4
-      (13.2 if Trilinos includes Kokkos).
+      (12.14.1 if Trilinos includes Kokkos).
       Deal.II is known to work with Trilinos up to 13.4. Other versions of
       Trilinos should work too but have not been tested prior to the
       release.

--- a/doc/news/changes/20221230DanielArndt
+++ b/doc/news/changes/20221230DanielArndt
@@ -1,4 +1,4 @@
-Updated: The minimum version for Trilinos has been bumped to 13.2 if Trilinos
+Updated: The minimum version for Trilinos has been bumped to 12.14.1 if Trilinos
 bundles Kokkkos.
 <br>
 (Daniel Arndt, 2022/12/30)

